### ISSUE TITLE
Router navigateByUrl documentation does not indicate that it ignores queryParams

### ISSUE THREAD

**Issue by Jaarx** [angular/angular#22611]
https://angular.io/api/router/Router#navigateByUrl

The above documentation gives no indication that it ignores the queryParams NavigationExtras value. There should be something that explicitly mentions NavigationExtras that do nothing.

**Comment by jasonaden** [comment]
`queryParams` is used when creating the target URL, as well as when using `queryParamsHandling` strategy of `merge`, as well as returned as the default for `routerLink`s.

Please file a new issue if there's a specific thing missing from the documentation.

**Comment by Jaarx** [comment]
That seems wrong if we parse through the code:

navigateByUrl itself at first only uses the URL as provided from the function call: https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L452

When it goes to scheduleNavigation, the extras is passed along but nothing is done in scheduleNavigation with extras except passed to this.navigations.next(...): https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L567

this.navigations is processed here: https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L518

which calls executeScheduledNavigation: https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L574

Notably at this point the only values from extras that this cares about is skipLocationChange and replaceUrl: https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L586

TL;DR at no point does a navigateByUrl extract any queryParams from the NavigationExtras
